synapse/handlers/_base.py

```diff
@@ -282,3 +284,41 @@ def log_failure(f):
         federation_handler.handle_new_event(
             event, destinations=destinations,
         )
+
+    @defer.inlineCallbacks
+    def maybe_kick_guest_users(self, event, current_state):
```

Comment:
Might be nice to split this into a ``maybe_kick_guest_users`` and a ``kick_guest_users`` where the latter just takes the current_state.